packages/astro/src/runtime/server/render/slot.ts

```diff
@@ -49,17 +50,21 @@ export async function renderSlotToString(
 ): Promise<string> {
 	let content = '';
 	let instructions: null | RenderInstruction[] = null;
-	let iterator = renderSlot(result, slotted, fallback);
-	for await (const chunk of iterator) {
-		if (typeof chunk.type === 'string') {
-			if (instructions === null) {
-				instructions = [];
+	const temporaryDestination: RenderDestination = {
+		write(chunk) {
+			if (chunk instanceof Response) return;
+			if (typeof chunk === 'object' && 'type' in chunk && typeof chunk.type === 'string') {
+				if (instructions === null) {
+					instructions = [];
+				}
+				instructions.push(chunk);
+			} else {
+				content += chunkToString(result, chunk);
 			}
-			instructions.push(chunk);
-		} else {
-			content += chunk;
-		}
-	}
+		},
+	};
+	const renderInstance = renderSlot(result, slotted, fallback);
+	await renderInstance.render(temporaryDestination);
```

Comment:
Slots are very tricky to render. They are split as `content` and `instructions`. `content` is the basic HTML, `instructions` are like astro island scripts/styles, and head content.

This out-of-order sorting makes it that we need a temporary destination to "collect" and sort the chunks.